ucum-java8/src/main/java/systems/uom/ucum/format/UCUMFormat.java

```diff
@@ -207,24 +207,48 @@ public Appendable format(Unit<?> unknownUnit, Appendable appendable)
 		    temp.append(')');
 		}
 		int pow = productUnits.get(u);
-		if (app.length() > 0) { // Not the first unit.
-		    if (pow >= 0) {
-			app.append('.');
-		    } else {
-			app.append('/');
-			pow = -pow;
-		    }
+				int indexToAppend;
+				if(app.length() > 0) { // Not the first unit.
+
+					if(pow >= 0) {
+
+						if(app.indexOf("1/") >= 0) {
+							indexToAppend = app.indexOf("1/");
```

Comment:
Why do you go to the trouble of calculating the index to append (actually this is the index to _insert_, not append) if you don't plan on using that value? i.e. You call the same method to calculate the same value over and over in the next line.

P.S. Isn't this an ideal place to use `final`?